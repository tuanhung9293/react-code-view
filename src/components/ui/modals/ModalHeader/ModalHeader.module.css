@value colors: 'styles/colors.css';
@value separator from colors;

.base {
  text-align: center;
  padding: 40px;
}

.withCloseButton {
  position: relative;
  padding: 0px 50px;
}

.closeButton {
  position: absolute;
  right: 10px;
  top: 50%;
  transform: translateY(-50%);
}