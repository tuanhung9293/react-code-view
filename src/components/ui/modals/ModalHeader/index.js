import React from 'react'
import classNames from 'classnames'
import ClearButton from 'components/ui/buttons/ClearButton'
import CrossIcon from 'components/ui/CrossIcon'
import styles from './ModalHeader.module.css'

function ModalHeader({heading, subheading, closeTo, onClose, children}) {
  const wrapperClassName = classNames({
    [styles.base]: true,
    [styles.withCloseButton]: closeTo || onClose
  })

  return (
    <header className={wrapperClassName}>
      <div>
        {heading && <h1>{heading}</h1>}
        {subheading && <p>{subheading}</p>}
        {children}
      </div>
      {(closeTo || onClose) && (
        <div className={styles.closeButton}>
          <ClearButton
            to={closeTo && closeTo}
            onClick={onClose && onClose}
          >
            <CrossIcon dark />
          </ClearButton>
        </div>
      )}
    </header>
  )
}

export default ModalHeader