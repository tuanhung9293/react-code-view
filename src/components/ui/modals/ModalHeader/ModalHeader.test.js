import React from 'react'
import {shallow} from 'enzyme'
import ModalHeader from './'

it('renders heading', () => {
  expect(shallow(
    <ModalHeader
      heading="Oh my head"
    />
  )).toMatchSnapshot()
})

it('renders subheading', () => {
  expect(shallow(
    <ModalHeader
      subheading="Oh my neck"
    />
  )).toMatchSnapshot()
})

it('renders with a close button if there is a closeTo param provided', () => {
  expect(shallow(
    <ModalHeader
      heading="I close with a link"
      closeTo="/the/url/to/close/to"
    />
  )).toMatchSnapshot()
})

it('renders with a close button if there is an onClose param provided', () => {
  expect(shallow(
    <ModalHeader
      heading="I close with a function"
      onClose={() => {}}
    />
  )).toMatchSnapshot()
})

it('calls the provided onClose function if the close button is clicked', () => {
  const closeHandler = jest.fn()
  const wrapper = shallow(
    <ModalHeader
      heading="I should react"
      onClose={closeHandler}
    />
  )
  wrapper.find('ClearButton').simulate('click')
  expect(closeHandler).toHaveBeenCalledTimes(1)
})