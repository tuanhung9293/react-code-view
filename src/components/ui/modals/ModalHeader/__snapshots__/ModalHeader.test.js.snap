// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heading 1`] = `
<header
  className="base"
>
  <div>
    <h1>
      Oh my head
    </h1>
  </div>
</header>
`;

exports[`renders subheading 1`] = `
<header
  className="base"
>
  <div>
    <p>
      Oh my neck
    </p>
  </div>
</header>
`;

exports[`renders with a close button if there is a closeTo param provided 1`] = `
<header
  className="base withCloseButton"
>
  <div>
    <h1>
      I close with a link
    </h1>
  </div>
  <div
    className="closeButton"
  >
    <ClearButton
      to="/the/url/to/close/to"
    >
      <CrossIcon
        dark={true}
      />
    </ClearButton>
  </div>
</header>
`;

exports[`renders with a close button if there is an onClose param provided 1`] = `
<header
  className="base withCloseButton"
>
  <div>
    <h1>
      I close with a function
    </h1>
  </div>
  <div
    className="closeButton"
  >
    <ClearButton
      onClick={[Function]}
    >
      <CrossIcon
        dark={true}
      />
    </ClearButton>
  </div>
</header>
`;
